app/uk/gov/hmrc/contactadvisors/controllers/EmailController.scala

```diff
@@ -35,15 +36,20 @@ class EmailController @Inject()(
   controllerComponents: MessagesControllerComponents,
   val authConnector: AuthConnector,
   val emailService: EmailService,
+  val conf: Configuration,
   val env: Environment)(implicit val appConfig: uk.gov.hmrc.contactadvisors.FrontendAppConfig)
     extends FrontendController(controllerComponents) with AuthorisedFunctions {
 
+  lazy val predicates: Option[Predicate] =
+    conf
+      .getOptional[Seq[String]](s"email.stride.roles")
+      .map(_.distinct)
+      .collect { case x :: xs => xs.foldLeft[Predicate](Enrolment(x))((b, a) => b or Enrolment(a)) }
```

Comment:
can you explain what/why you are doing this here :)?

Reply:
I want to construct Option[Predicate] so that I can use  it with "and" with AuthProviders if it is Some(List(predicates)) or AuthProvider alone if it is None

Reply:
i meant what is the business logic here :)

Reply:
this is traversing email.stride.roles string array and creating a predicate list with the Auth library "or". So it will look something like Enrolement("ROLE_1") or Enrolement("ROLE2") if the string array email.stride.roles = ["ROLE_1", "ROLE_1", "ROLE_2"]